src/pages/guides/tutorials/stats-addon.md

```diff
@@ -0,0 +1,1108 @@
+
+// We can call this method now
+await sandboxProxy.drawRect();
+```
+
+#### Document Sandbox
+
+```js
+import addOnSandboxSdk from "add-on-sdk-document-sandbox";
+const { runtime } = addOnSandboxSdk.instance;
+
+// exporting a payload to the iframe
+runtime.exposeApi({ 
+  drawRect: () => {
+    // uss the Document API to draw a Rectangle shape...
+  }
+});
+```
+
+The following diagram helps visualize the process.
+
+![Proxy objects](images/stats-addon-proxy.png)
+
+### Proxy API
+
+Now that we've seen how contexts expose and import each other's API, let's discuss what's inside the payload objects that cross this environment boundary.
+
+#### Functions
+
+Most of the time, you're going to expose **functions**.
+
+```js
+// Document Sandbox
+runtime.exposeApi({
+  drawRect:    () => { /* ... */ }, // 👈
+  drawEllipse: () => { /* ... */ }, // 👈
+  drawShape:   () => { /* ... */ }, // 👈
+  // etc.
+});
+```
+
+In the above example, if `drawShape()` needs to call either `drawEllipse()` or `drawRect()`, you may use the *method shorthand syntax*.
+
+```js
+// Document Sandbox
+runtime.exposeApi({
+  drawRect()    { /* ... */ }, // 👈
+  drawEllipse() { /* ... */ }, // 👈
+  drawShape() { 
+    this.drawRect();           // 👆
+    // or
+    this.drawEllipse();        // 👆
+  },
+  // etc.
+});
+```
+
+When not strictly necessary, keep functions private and expose only the ones needed.
+
+```js
+// Document Sandbox
+const drawRect    = () => { /* ... */ };  // 👈 private
+const drawEllipse = () => { /* ... */ };  // 👈 private
+
+runtime.exposeApi({
+  drawShape() { 
+    drawRect();    // 👆
+    // or
+    drawEllipse(); // 👆
+  }
+});
+```
+
+Here, `drawRect()` and `drawEllipse()` exist within the closure of the `drawShape()` function exposed to the iframe and will work just fine when the iframe invokes it. This notion of "private" variables defined in one context can be exploited in various ways, for instance, with a counter as follows.
+
+<CodeBlock slots="heading, code" repeat="2" languages="iframe, Document Sandbox"/>
+
+#### UI iframe
+
+```js
+const sandboxProxy = await runtime.apiProxy("documentSandbox");
+
+// Calling the exposed method
+const shapesNo = await sandboxProxy.drawShape();
+console.log("Shapes drawn", shapesNo);
+```
+
+#### Document Sandbox
+
+```js
+const drawRect    = () => { /* ... */ };
+const drawEllipse = () => { /* ... */ };
+
+let counter = 0;    // 👈 private to the sandbox
+
+runtime.exposeApi({
+  drawShape() { 
+    if (counter >= 10) { throw new Error("Shape's budget depleted!"); }
+    // Randomly draw either a rectangle or an ellipse
+    Math.random() < 0.5 ? drawRect() : drawEllipse();
+
+	return ++counter; // 👆
+  }
+});
+```
+
+The `counter` variable in the Document Sandbox persists between iframe calls and can be accessed by `drawShape()`; in this scenario, it's returned so the iframe knows the number of shapes created so far. This brings us to the concept of **returned values**.
```

Comment:
May be worth pointing out that `counter` is persisted for the user's session in the document, and is not shared w/ anyone else or any other document the user may have open.

Reply:
Added.